[WIP] Fix lint/useless-access-modifier offenses

nickgnd opened the pull request:
Realated to the issue #12034 

**Fix lint/useless-access-modifier offenses raised by Rubocop.**

in the file [`/gems/pending/util/miq-password.rb:`](https://github.com/ManageIQ/manageiq/blob/master/gems/pending/util/miq-password.rb#L177)
there were two class methods declared as `protected` and identified as offenses by Rubocop. 
I set this 2 methods public. Is ok for you?
## Steps for Testing/QA [Optional]

`rubocop --only Lint/UselessAccessModifier`
## Links

See also #11031 and #11032

jrafanie commented on app/models/manageiq/providers/openstack/network_manager/cloud_subnet.rb:
```diff
@@ -55,13 +55,12 @@ def with_provider_object
     connection_options[:tenant_name] = cloud_tenant.name if cloud_tenant
     connection_options
   end
+  private_class_method :connection_options
```
I could be wrong though.  Either way, I think connection_options was changed over there in a different file and might be public.  Please see the other PR and let me know what you think.

nickgnd replied:
I checked the other PR (#12041) and I didn't notice changes to this file: app/models/manageiq/providers/openstack/network_manager/cloud_subnet.rb

So, for the moment I keep my changes, let me know if is ok for you.

jrafanie commented on app/models/mixins/supports_feature_mixin.rb:
```diff
@@ -222,5 +220,9 @@ def define_supports_feature_methods(feature, is_supported: true, reason: nil, &b
+
+    private :unsupported
+    private :unsupported_reason_add
+    private :define_supports_feature_methods
```
Typo maybe?  private_class_method

jrafanie commented on gems/pending/VolumeManager/LVM/scanner.rb:
```diff
@@ -86,14 +86,13 @@ def self.labelScan(d)
   end # def self.labelScan
 
-  private
-
```
Sorry, I think there's overlap with #12041 in this file.  I'm sorry, I didn't realize the two different COPs would point out some of the same code.

jrafanie commented on gems/pending/disk/modules/MSCommon.rb:
```diff
@@ -256,27 +257,32 @@ def self.process_bae
       @bae << @file.read(BAE_SIZE).unpack('N')[0]
     end
   end
+  private_class_method :process_bae
```
Same as above, sorry.

jrafanie commented on gems/pending/util/miq-password.rb:
```diff
@@ -174,7 +174,7 @@ def self.generate_symmetric(filename = nil)
   end
 
-  protected
+  # protected
```
Overlap with the other PR, I think.  I asked another contributor on that other PR if this protected was needed because protected on class/singleton methods is useless.

jrafanie commented on gems/pending/util/miq_apache/miq_apache.rb:
```diff
@@ -149,6 +147,7 @@ def self.run_apache_cmd(command)
         $log.warn("MIQ(MiqApache::Control.run_apache_cmd) Apache command #{command} with result: #{res} failed with error: #{err}") if $log
       end
     end
+    private_class_method :run_apache_cmd
```
Overlap with the other PR, sorry.

jrafanie commented on gems/pending/util/miq_logger_processor.rb:
```diff
@@ -122,7 +122,7 @@ def cleanup_get_next_line
 class MiqLoggerLine < String
   def parts
-    @parts ||= self.class.split_raw_line(self).freeze
+    @parts ||= self.class.send(:split_raw_line, self).freeze
```
Overlap again.  Sorry.  I don't know that it makes sense to make this really private.  If we're calling the class method from an instance, it's a public method.  I'd prefer to just make it public by removing the private line.  cc @Fryguy

Fryguy replied:
Yeah, in this case make it either protected or just leave it public.

jrafanie replied:
We're adding a TODO in the other PR.  In a followup PR, we can drop the comment and just keep it public so there's no confusion.

https://github.com/ManageIQ/manageiq/pull/12041/files#diff-037377d58165f2739a6b9aeb4ef6976fR140

jrafanie commented on gems/pending/util/xml/miq_rexml.rb:
```diff
@@ -83,6 +81,7 @@ def self.findRegElementInt(paths, ele)
       return ele
     end
   end
+  private_class_method :findRegElementInt
```
Ugh, sorry, overlap with the other PR.

jrafanie commented on lib/vmdb/loggers.rb:
```diff
@@ -91,5 +90,6 @@ def self.apply_config_value_logged(config, logger, level_method, key)
         logger.send("#{level_method}=", new_level)
       end
     end
+    private_class_method :apply_config_value_logged
```
Overlap, 😢

jrafanie commented:
@nickgnd Thanks for your contribution.  Can you remove the same changes from that other PR and squash your similar commits?  If the commits stand alone, feel free to keep them.

nickgnd commented:
@jrafanie ok! I work on this in the next days!

nickgnd commented:
Hi @jrafanie, sorry for my inactivity, I was very busy in the past days. 

I rebase my branch with the master one and all the files under `gems` folder was deleted, so I'm going to delete these files too including my changes for this PR. 

```
deleted by us:   gems/pending/VolumeManager/LVM/scanner.rb
	deleted by us:   gems/pending/disk/modules/MSCommon.rb
	deleted by us:   gems/pending/fs/ReiserFS/file_data.rb
	deleted by us:   gems/pending/spec/util/miq_apache/control_spec.rb
	deleted by us:   gems/pending/util/miq-password.rb
	deleted by us:   gems/pending/util/miq_apache/miq_apache.rb
	deleted by us:   gems/pending/util/miq_logger_processor.rb
	deleted by us:   gems/pending/util/xml/miq_rexml.rb
```

nickgnd commented:
I rebased my branch and squashed all commits into [one](https://github.com/nickgnd/manageiq/commit/7888df450e7b739d62a36064d8122757bfc707f1).

**As requested I revert all the edits on files which will be touched by PR #12041**

Now, if I run `rubocop --only Lint/UselessAccessModifier` it detects one warning on file `lib/vmdb/loggers.rb`.

This offences will be fixed on PR  #12041 ([detail](https://github.com/ManageIQ/manageiq/pull/12041/files#diff-f2c7eefe837eb1f96c1b4ceb83e3ee97)) 

```
Offenses:

lib/vmdb/loggers.rb:40:5: W: Useless private access modifier.
    private
    ^^^^^^^

5201 files inspected, 1 offense detected
```

chrisarcand commented:
Closing this as it's quite stale and there undoubtedly are too many conflicts to not need to start fresh anyway. Please open a new PR if you still wish to take this on. 💙
